lib/poller.js

```diff
@@ -0,0 +1,51 @@
+'use strict'
+
+const { EventEmitter } = require('events')
+
+class Poller extends EventEmitter {
+  constructor ({
+    pollingRateSeconds,
+    serializeSegments,
+    splitioApi
+  }) {
+    super()
+    this.cache = {}
+    this.pollingRateSeconds = pollingRateSeconds || 300
+    this.serializeSegments = serializeSegments || false
+    this.splitioApi = splitioApi
+    this.timeout = null
+  }
+
+  async pollForChanges () {
+    this.on('poll', async () => {
```

Comment:
why the `this.on`?  just remove this and call it directly from the interval timer.